AppCenter/AppCenterTests/MSUserDefaultsTests.m

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAppCenterUserDefaults.h"
+#import "MSAppCenterUserDefaultsPrivate.h"
+#import "MSLoggerInternal.h"
+#import "MSTestFrameworks.h"
+#import "MSUtility.h"
+#import "MSWrapperLogger.h"
+
+@interface MSUserDefaultsTests : XCTestCase
+
+@end
+
+static NSString *const kMSAppCenterUserDefaultsMigratedKey = @"MSAppCenter310AppCenterUserDefaultsMigratedKey";
+
+@implementation MSUserDefaultsTests
+
+- (void)setUp {
+  NSArray *userDefaultKeys = [[[NSUserDefaults standardUserDefaults] dictionaryRepresentation] allKeys];
+  for (NSString *key in userDefaultKeys) {
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:key];
+  }
```

Comment:
Does [`removePersistentDomain`](https://developer.apple.com/documentation/foundation/userdefaults/1417339-removepersistentdomain) work here?

Reply:
I don't think so. we need to remove keys without prefix, too.